### DISCUSSION TITLE
screen tearing while moving node

### DISCUSSION THREAD

**Discussion by chadwyk** [axmolengine/axmol#2117] [General]
I am making a MacOS application with Axmol 2.1.4. I have an issue with  unwanted "screen tearing effect" when moving a node.  It was originally happening on a parallax node and I thought it had to do with large images on it, but I stripped it down to just the very basic c++ new Hello World project and added code in the update to move the Axmol logo using the keys and it has the issue as well.

Any ideas? This is how I am moving the node.

`void MainScene::update(float delta)
{
           Vec2 newPosition;
           // Move the camera based on arrow key presses when not in test mode
           Vec2 moveDirection(0, 0);
           float moveSpeed = 200.0f * delta; // Adjust speed as needed, scaled by delta time
           if (_isLeftPressed)
               moveDirection.x += moveSpeed;
           if (_isRightPressed)
               moveDirection.x -= moveSpeed;
           if (_isUpPressed)
               moveDirection.y -= moveSpeed;
           if (_isDownPressed)
               moveDirection.y += moveSpeed;

           newPosition = sprite->getPosition() + moveDirection;

    sprite->setPosition(newPosition);
}`

**Comment by chadwyk** [comment]
Hmmmm... I decided to compile and run the Axmol cpp tests. I can see it happening on the tests as well, like the actions test. When actions are run on the three sprites, and they move quickly, they have a screen tearing effect occasionally.

**Comment by chadwyk** [comment]
It doesn't seem to do it on the iOS simulator as far as I can tell.... doing more testing

**Comment by chadwyk** [comment]
I found the fix after searching all day: https://github.com/axmolengine/axmol/discussions/1684  

It looks like you have to change vsync = true in the engine's GLViewImpl.cpp which worked for me. 

But a better way is to edit the AppDelegate, and after you set the glContextAttrs, add the below code.
Maybe it could be added to the default project template in the next version??? 

```
glContextAttrs.vsync = true;
GLView::setGLContextAttrs(glContextAttrs); 
```

**Reply by halx99** [reply]
sure, refer to https://github.com/axmolengine/axmol/pull/2118
